### ISSUE TITLE
Database Resource Owner login function fails, calling a non-existent Auth0 endpoint

### ISSUE THREAD

**Issue by mildmojo** [auth0/auth0-python#251]
### Describe the problem

The Python SDK fails to perform Resource Owner Password Flow database logins because it calls the non-existent endpoint `/auth/ro`. The Auth0 API docs say [this legacy endpoint is deprecated](https://auth0.com/docs/api/authentication#database-ad-ldap-active-) (apparently now removed), and clients [should use](https://auth0.com/docs/api/authentication#resource-owner-password) `/oauth/token` instead.

### What was the expected behavior?

I expected that calling `.login()` on a database auth client would call the Auth0 API and return an auth token.

### Reproduction

In the Auth0 admin dashboard, create a new database connection, create a new application (regular web app) that uses that connection. Create a user in that database or go through the web-based signup flow, then attempt to log in using the Python SDK.

```
# testauth.py
from auth0.v3.authentication import Database as Auth0Database

auth0_client = Auth0Database('DOMAIN')
login_result = auth0_client.login('CLIENT_ID', 'EMAIL_ADDRESS', 'PASSWORD', 'CONNECTION_ID')
print(login_result)
```

This fails with an HTTP "404: Not Found" error because Auth0's `/auth/ro` endpoint no longer exists.

```
Traceback (most recent call last):
  File "testauth.py", line 5, in <module>
    login_result = auth0_client.login(client_id, email, pw, connection_id)
  File "/usr/local/lib/python3.7/dist-packages/auth0/v3/authentication/database.py", line 37, in login
    return self.post('https://{}/oauth/ro'.format(self.domain), data=body)
  File "/usr/local/lib/python3.7/dist-packages/auth0/v3/authentication/base.py", line 49, in post
    return self._process_response(response)
  File "/usr/local/lib/python3.7/dist-packages/auth0/v3/authentication/base.py", line 58, in _process_response
    return self._parse(response).content()
  File "/usr/local/lib/python3.7/dist-packages/auth0/v3/authentication/base.py", line 85, in content
    message=self._error_message())
auth0.v3.exceptions.Auth0Error: 404: Not Found
```

### Environment

Python v3.7.5
auth0-python v3.14.0

**Comment by lbalmaceda** [comment]
@mildmojo  I cannot find information or announcements on whether this was removed or not. I know it was deprecated in 2017 according to [this entry](https://auth0.com/docs/product-lifecycle/deprecations-and-migrations#resource-owner-password-oauth-ro-deprecation). The recommendation is to migrate to `/oauth/token` and use the password realm grant. We already support that endpoint from [this method](https://github.com/auth0/auth0-python/blob/master/auth0/v3/authentication/get_token.py#L117). 

Did you read it somewhere that you had to use that method? How can we improve the docs? Thanks.

In general, legacy grant types such as RO are still available for old applications/tenants to help migrate, but new ones don't have it available at all. If this is a new application you are setting up, you should not be using the deprecated endpoint.

**Comment by mildmojo** [comment]
@lbalmaceda I think the `/auth/ro` endpoint is gone for me because I'm setting up a new application, as you suggest, and the SDK method still exists for legacy applications which may still have access to the deprecated endpoint?

The only docs I found for this python SDK were in this repo's README (which often links to auth0.com docs). In the "Authentication SDK Usage" section, there's an example of using the Social login strategy. Since my Auth0 app is backed with a database connection, I assumed I had to swap out Social for Database in the example. I didn't think to look at GetToken, and there weren't any examples of using GetToken for regular app authentication in the README, just Auth0's Management API.

I've had a vast amount of trouble understanding the Auth0 docs generally, and it's been very difficult to connect Auth0 admin panel items, high level docs (auth0.com/docs/flows), HTTP API docs (auth0.com/docs/api), and this SDK's documentation together. I'd hoped to find docs for this SDK's classes and functions, with a lot of practical examples. One of the reasons I went looking for an SDK was because I was struggling to interpret the Auth0 docs which were light on code examples ("okay but how do I actually do this"), and I hoped the SDK would make more sense.

Thanks for explaining here; I see now that I could've used GetToken.login() for Resource Owner Password Flow logins against `/oauth/token`. :+1: I ended up switching to `requests` for my integration.
